### DISCUSSION TITLE
metaMDS issue

### DISCUSSION THREAD

**Discussion by tuulri** [vegandevs/vegan#717] [Q&A]
Hi,

I´m using vegan::metaMDS() to perform NMDS analysis. My data consist of presence/absence records of 568 species in 10 000 sites. Sites with no species present are excluded. I have tried using several different 'k', 'trymax' and 'maxit' parameters but each try ends up with ”No convergence monoMDS stopping criteria: scale factor of the gradient < sfgrmin". How this should be interpreted? Stress value between runs remains quite constant, though some runs get notably lower values than others. Example of the latest try is shown below.

nmds1 <- metaMDS(species data, distance="jaccard", k=6, trymax=100, maxit=500 )
...
Run 92 stress 0.2114321 
Run 93 stress 0.05507131 
... Procrustes: rmse 0.001076182  max resid 0.02081336 
Run 94 stress 0.06146071 
Run 95 stress 0.2115075 
...
*** No convergence -- monoMDS stopping criteria:
   100: scale factor of the gradient < sfgrmin

I can´t figure out what I should do differently to get metaMDS to converge. Or could the problem be caused e.g. by the size of my data? Any help is much appreciated!

Best regards,
Tuuli

**Comment by jarioksa** [comment]
Firstly, `k=6` may be too high: wildly varying stresses (from 0.2 to 0.06) hint to problems in weaving the manifold in high-dimensional space. Try with `k=3` or `k=4` (I very rarely need to go to 4-dim space).

Then about stopping criteria: these are passed to `monoMDS` which says:


> smin,` sfgrmin, sratmax: Convergence criteria: iterations stop when
          stress drops below ‘smin’, scale factor of the gradient drops
          below ‘sfgrmin’, or stress ratio between two iterations goes
          over ‘sratmax’ (but is still < 1).

And the default call is:
```r
monoMDS(dist, y, k = 2, model = c("global", "local", "linear", "hybrid"),
         threshold = 0.8, maxit = 200, weakties = TRUE, stress = 1,
         scaling = TRUE, pc = TRUE, smin = 1e-4, sfgrmin = 1e-7,
         sratmax=0.999999, ...)
```
The default seems to be `sfgrmin = 1e-7`. Try something lower, such as `sfgrmin = 1e-9`  in the `metaMDS`  call. After running the cycle, `metaMDS` will report the counts for stopping criteria and you can see if something else should be made stricter. Also save the result and use it as a starting solution for the next round (`previous.best=`): it may be that you already had a solutions, but it was not detected yet.

**Comment by tuulri** [comment]
Thank you @jarioksa, 

Lower sfgrmin value helped to decrease the stress level and enabled using fewer dimensions (k=4). However, the next issue I´m running into is the stratmax value (No convergence: stressratio > stratmax). In the Convergence criteria it says 'iterations stop when ... or stress ratio between two iterations goes over ‘sratmax’ (but is still < 1)' and the default value is 0.999999´ and 'stratmax: Change in stress. Values close to one mean almost unchanged stress. This may mean a solution, but it can also signal stranding on suboptimal solution with flat stress surface.'. Between my runs, the stress indeed is almost unchanging and I´m not sure, how I should interpret the situation. Should I try setting a lower 'stratmax' value or change other parameters in order to get more varying stress levels? (In tests I have run, I have used smaller 'trymax' value in order to get the analysis run quicker.)

nmds5 <- metaMDS(spe, distance="jaccard", k=4, trymax=20, maxit=600, previous.best = nmds4, 
                sfgrmin = 1e-12, stratmax=0.99999999) 
...
Run 18 stress 0.08476893 
... Procrustes: rmse 0.001074971  max resid 0.02233631 
Run 19 stress 0.08488367 
... Procrustes: rmse 0.0008033794  max resid 0.02244835 
Run 20 stress 0.08486097 
... Procrustes: rmse 0.001116907  max resid 0.02236367 
*** No convergence -- monoMDS stopping criteria:
    20: stress ratio > sratmax

Many thanks again!

**Comment by tuulri** [comment]
Additional note about the data. Species presence/absence records are based on modelled species distributions due to which there might be less noise than e.g. in observations from field data. I don´t know, if the nature of the data affects the stress levels and working of NMDS analysis.

Best regards,
Tuuli

**Comment by charlinegiguet** [comment]
Hi, 
I met the same problem on Environmental DNA data. My data are numbers of DNA sequences for each species (151 in total) and sample replicate (24 sample * 9 PCR replicates).
I tried to perform an NMDS analysis  but don't get convergence. I tested several values for stratmax or sfgrmin and sometimes, I manage to obtain a convergence but when I re-try it doesn't work again. so the convergence is not stable.

Because the replication with the PCR can be stochastic when the DNA content at the beginning is low, I pooled replicates to finally only have 3 replicated values by samples. Doing that I managed to reach convergence, but I don't know really how to explain and justify the choice. 
Is someone can help me ?
thanks !

**Comment by jarioksa** [comment]
What do you mean with "... the convergence is not stable"? 

The convergence is not stable if `metaMDS` reports convergences in separate runs, and these claimed convergences differ in `procrustes()` analysis (this can happen). If you do not get a convergence in some run, then that run failed to find the convergence, but it has nothing to do with the stability of the convergence.

How many random starts have you used in your tests?

**Comment by charlinegiguet** [comment]
Hi Jari,

Thanks for your explanation.
I asked for 100 starts.

My last command was:
AIG_NMDS <- metaMDS(aig[,3:152], distance = "bray », k=3, trymax=100, maxit = 2000, smin = 1e-4, sfgrmin = 1e-11, sratmax= 0.99999)

Thanks
Charline

**Comment by MonaFuhrmann** [comment]
Hei,
I see this issue is still open - I have a similar problem, getting the warning after a call to metaMDS
>metaMDS(dis_bray.curtis,k=3,autotransform=F,trymax=100)

Run.....1-100

Best solution was not repeated -- monoMDS stopping criteria:  
6: no. of iterations >= maxit 
4: stress ratio > sratmax"

My stress is 0.14, so ok. I just wonder how to interpret this warnings and if I can still trust the results (stressplot is not great, as I have many sites with no shared species).
